lib/sles4sap/ipaddr2.pm

```diff
@@ -1915,4 +1946,244 @@ sub ipaddr2_test_other_vm {
     }
 }
 
+=head2 ipaddr2_clean_network_peering
+
+    ipaddr2_clean_network_peering();
+
+Cleanup the network peering if needed.
+
+=over
+
+=back
+
+=cut
+
+sub ipaddr2_clean_network_peering {
+    return unless get_var("IBSM_RG");
+    my $source_group = ipaddr2_azure_resource_group();
+    my $ibs_mirror_resource_group = get_required_var('IBSM_RG');
+
+    qesap_az_vnet_peering_delete(source_group => $source_group, target_group => $ibs_mirror_resource_group);
+}
+
+=head2 ipaddr2_network_peering
+
+    ipaddr2_network_peering();
+
+Create network peering
+
+=over
+
+=back
+
+=cut
+
+sub ipaddr2_network_peering {
+    my $ibs_mirror_resource_group = get_required_var('IBSM_RG');
+
+    # remove the older peering
+    my $vnet_name = qesap_az_get_vnet($ibs_mirror_resource_group);
+    qesap_az_clean_old_peerings(rg => $ibs_mirror_resource_group, vnet => $vnet_name);
+
+    my $rg = ipaddr2_azure_resource_group();
+    qesap_az_vnet_peering(source_group => $rg, target_group => $ibs_mirror_resource_group);
+}
+
+=head2 ipaddr2_add_server_repos_to_hosts
+
+    ipaddr2_add_server_repos_to_hosts();
+
+Add download.suse.de server to hosts by specifying IBSM address
+
+=over
+
+=back
+
+=cut
+
+sub ipaddr2_add_server_repos_to_hosts {
+    my $ibsm_ip = get_required_var('IBSM_IP');
+    my $bastion_pubip = ipaddr2_bastion_pubip();
+    foreach my $id (1 .. 2) {
+        ipaddr2_ssh_internal(id => $id,
+            cmd => "echo \"$ibsm_ip download.suse.de\" | sudo tee -a /etc/hosts",
+            bastion_ip => $bastion_pubip);
+    }
+
+    # Add repos
+    my $count = 0;
+    my @repos = split(/,/, get_var('INCIDENT_REPO', ''));
+    while (defined(my $maintrepo = shift @repos)) {
+        next if $maintrepo =~ /^\s*$/;
+        if ($maintrepo =~ /Development-Tools/ or $maintrepo =~ /Desktop-Applications/) {
+            record_info("MISSING REPOS", "There are repos in this incident, that are not uploaded to IBSM. ($maintrepo). Later errors, if they occur, may be due to these.");
+            next;
+        }
+        my $zypper_cmd = "sudo zypper --no-gpg-checks ar -f -n TEST_$count";
+        $zypper_cmd = $zypper_cmd . " --priority " . get_var('REPO_PRIORITY') if get_var('REPO_PRIORITY');
+        $zypper_cmd = $zypper_cmd . " $maintrepo TEST_$count";
+
+        foreach my $id (1 .. 2) {
+            ipaddr2_ssh_internal(id => $id,
+                cmd => $zypper_cmd,
+                bastion_ip => $bastion_pubip);
+        }
+        $count++;
+    }
+}
+
+=head2 ipaddr2_patch_system
+
+    ipaddr2_patch_system();
+
+Patch system
+
+=over
+
+=back
+
+=cut
+
+sub ipaddr2_patch_system {
+    my $bastion_pubip = ipaddr2_bastion_pubip();
```

Comment:
Please use pattern proposed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20882/files#r1907006023